modules/buildkite-agent.nix

```diff
@@ -0,0 +1,114 @@
+{ config, lib, pkgs, ... }:
```

Comment:
I don't see what this file changes fundamentally that's needed for buildkite to work. All buildkite specific NixOS options already exist upstream.

Reply:
As discussed on the meeting, this allows provisioning secrets, without committing them to the Nix source.

Reply:
That's not required or important for the prototype issue, if I understand correctly.

Reply:
We've addressed this in on Slack: this is needed to avoid leaking secrets.